xml/map_test.go

```diff
@@ -0,0 +1,78 @@
+package xml
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestWrappedMap(t *testing.T) {
+	buffer := bytes.NewBuffer(nil)
+	scratch := make([]byte, 64)
+
+	func() {
+		root := StartElement{Name: Name{Local: "map"}}
+		m := newMap(buffer, &scratch, root, false)
```

Comment:
This highlights why having optional parameters would be confusing Reading this code I'd expect the wrapping tag to be `map`, but `map` is not included in the encoded output.  This is the reason for splitting the map constructors.